### PR TITLE
Copilot/resolve hustlecodex issue

### LINKED ISSUE #1: hustlecodeX

### PR DESCRIPTION
**Pull request by DubjamMusic**

### CODE REVIEW

File: agent.py [hunk 1]
```diff
@@ -47,28 +83,45 @@ def chat(self, user_message: str) -> str:
...
+                # Handle OpenAI API errors (check specific types if available)
+                if (APIError is not None and APIConnectionError is not None and 
+                    RateLimitError is not None and 
+                    isinstance(e, (APIError, APIConnectionError, RateLimitError))):
+                    error_msg = f"OpenAI API error: {e}"
+                else:
+                    error_msg = f"Unexpected error: {e}"
+                print(error_msg)
+                response_text = f"Agent received: {user_message} (error occurred)"
+                self.add_message("assistant", response_text)
```
[the remark this answers is not in the document]
Comment (DubjamMusic) [reply agent.py:112]: You are assisting on the HustleCodex project.
Your role is to help incrementally harden a real OpenAI-powered GPT agent into production-ready infrastructure.
Do not suggest speculative features unless explicitly asked.
Project rules:
Preserve working behavior at all times
Prefer small, reviewable changes
Avoid abstractions until they are explicitly required
Never remove functionality without replacement
Current priorities (in order):
Stabilize the OpenAI-backed agent as a known-good baseline
Add minimal smoke tests to validate API integration
Introduce a thin provider interface to decouple OpenAI-specific logic
Centralize configuration (model, temperature, system prompt, API key source)
Improve logging and error messages for real-world debugging
Agent behavior constraints:
System prompts must persist across resets
Message roles must be validated before API calls
Missing API keys or dependencies must fail loudly and clearly
No tool/function calling unless explicitly implemented
No long-term memory or persistence unless requested
Code style expectations:
Clear, readable Python over clever abstractions
Explicit error handling over silent failures
Comments explain why, not what
Follow existing project structure unless improving it deliberately
When reviewing or generating code:
Prefer edits that align with existing architecture
Point out risks, edge cases, and hidden assumptions
Suggest tests when behavior changes
Flag scope creep immediately
If uncertain, ask for clarification rather than guessing.

File: agent.py [hunk 2]
```diff
@@ -12,27 +12,63 @@
...
             api_key: OpenAI API key (if not provided, will look for OPENAI_API_KEY env var)
+            system_prompt: Optional system prompt to configure agent behavior
         """
         self.model = model
         self.api_key = api_key or os.getenv("OPENAI_API_KEY")
         self.conversation_history: List[Dict[str, str]] = []
+        self.client = None
+        
+        # Initialize conversation history with system prompt if provided
+        if system_prompt:
```
[the remark this answers is not in the document]
Comment (DubjamMusic) [reply agent.py:50]: > The system prompt is not validated before being added to conversation history. If system_prompt is an empty string or contains only whitespace, it will still be added as a system message. Consider adding validation to ensure the system_prompt is non-empty and meaningful before adding it to the conversation history.
> ```suggestion
>         # Initialize conversation history with system prompt if provided and meaningful
>         if system_prompt and system_prompt.strip():
> ```
